base/linalg/lapack.jl

```diff
@@ -3850,10 +3850,36 @@ for (gees, gges, elty, relty) in
     end
 end
 # Reorder Schur forms
-for (trsen, tgsen, elty) in
-    ((:dtrsen_, :dtgsen_, :Float64),
-     (:strsen_, :stgsen_, :Float32))
+for (trexc, trsen, tgsen, elty) in
+    ((:dtrexc_, :dtrsen_, :dtgsen_, :Float64),
+     (:strexc_, :strsen_, :stgsen_, :Float32))
     @eval begin
+        function trexc!(ifst::BlasInt, ilst::BlasInt, T::StridedMatrix{$elty}, Q::StridedMatrix{$elty})
+# *     .. Scalar Arguments ..
+#       CHARACTER          COMPQ
+#       INTEGER            IFST, ILST, INFO, LDQ, LDT, N
+# *     ..
+# *     .. Array Arguments ..
+#       DOUBLE PRECISION   Q( LDQ, * ), T( LDT, * ), WORK( * )
+            chkstride1(T, Q)
+            n = chksquare(T)
+            ldt = max(1, stride(T, 2))
+            ldq = max(1, stride(Q, 2))
+            work = Array($elty, 1)
```

Comment:
I think it should have length `n`. See http://www.netlib.org/lapack/explore-html/dc/d51/dtrexc_8f.html#a321d3f3a5e2768676a4d8ec2aa2ff2fe